test/gtfs_realtime_viz_test.exs

```diff
@@ -35,9 +35,246 @@ defmodule GTFSRealtimeVizTest do
+          }
+        }
+      ]
+    }
+
+    raw = Proto.FeedMessage.encode(data)
+
+    GTFSRealtimeViz.new_message(:test, raw, "this is the test data")
+    viz = GTFSRealtimeViz.visualize(:test, %{"First Route" => [["FR Only Stop", "this_is_the_stop_id", "124"]], "Second Route" => [["SR Only Stop", "125", "126"]]})
+
+    assert viz =~ "this is the test data"
+    assert viz =~ "this_is_the_vehicle_id"
+    assert viz =~ "First Route"
+    assert viz =~ "FR Only Stop"
+    assert viz =~ "Second Route"
+    assert viz =~ "SR Only Stop"
+  end
+
+  test "Only displays the given routes, even if there is data about other routes" do
+    data = %Proto.FeedMessage{
+      header: %Proto.FeedHeader{
+        gtfs_realtime_version: "1.0",
+      },
+      entity: [
+        %Proto.FeedEntity{
+          id: "123",
+          is_deleted: false,
+          vehicle: %Proto.VehiclePosition{
+            trip: %Proto.TripDescriptor{
+              trip_id: "this_is_the_trip_id",
+              route_id: "this_is_the_route_id",
+              direction_id: 0,
+            },
+            vehicle: %Proto.VehicleDescriptor{
+              id: "this_is_the_vehicle_id",
+              label: "this_is_the_vehicle_label",
+            },
+            position: %Proto.Position{
+              latitude: 0.00,
+              longitude: 0.00,
+            },
+            stop_id: "this_is_the_stop_id",
+          }
+        },
+        %Proto.FeedEntity{
+          id: "124",
+          is_deleted: false,
+          vehicle: %Proto.VehiclePosition{
+            trip: %Proto.TripDescriptor{
+              trip_id: "secondary_trip_id",
+              route_id: "other_route",
+              direction_id: 0,
+            },
+            vehicle: %Proto.VehicleDescriptor{
+              id: "different_vehicle",
+              label: "different_vehicle_label",
+            },
+            position: %Proto.Position{
+              latitude: 1.00,
+              longitude: 1.00,
+            },
+            stop_id: "126",
+          }
+        }
+      ]
+    }
+
+    raw = Proto.FeedMessage.encode(data)
+
+    GTFSRealtimeViz.new_message(:test, raw, "this is the test data")
+    viz = GTFSRealtimeViz.visualize(:test, %{"First Route" => [["FR Only Stop", "this_is_the_stop_id", "124"]]})
+
+    refute viz =~ "other_route"
+    refute viz =~ "different_vehicle"
+    assert viz =~ "this_is_the_vehicle_id"
+  end
+
+  describe "locations_we_care_about/1" do
+    test "turns a map of route name -> list of lists describing the stops into a list of all stops on all routes" do
+      routes = %{"First Route" => [["FR Only Stop", "123", "124"]], "Second Route" => [["SR First Stop", "125", "126"], ["SR Second Stop", "234", "432"]]}
+      assert GTFSRealtimeViz.locations_we_care_about(routes) == ["FR Only Stop", "123", "124", "SR First Stop", "125", "126", "SR Second Stop", "234", "432"]
+    end
+  end
+
+  describe "vehicles_we_care_about/2" do
```

Comment:
this test length got out of hand - does it seem worth it to move the test value out of the test and into either a function or a global variable?

Reply:
I think moving it into a function (maybe a factory?) could be good. You could check out the [test/support/](https://github.com/mbta/gtfs_realtime_viz/blob/4d4f324561fa1c3bca836ea90c44decaf6955361/test/support/data_helpers.ex) that the app has now, and maybe add to it.